backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/search/query/impl/SearchBackendContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.elasticsearch.search.query.impl;
+
+import java.util.List;
+import java.util.Set;
+
+import org.hibernate.search.v6poc.backend.elasticsearch.client.impl.URLEncodedString;
+import org.hibernate.search.v6poc.backend.elasticsearch.impl.MultiTenancyStrategy;
+import org.hibernate.search.v6poc.backend.elasticsearch.orchestration.impl.ElasticsearchWorkOrchestrator;
+import org.hibernate.search.v6poc.backend.elasticsearch.work.impl.ElasticsearchWorkFactory;
+import org.hibernate.search.v6poc.backend.spi.BackendImplementor;
+import org.hibernate.search.v6poc.engine.spi.SessionContext;
+import org.hibernate.search.v6poc.search.query.spi.HitAggregator;
+
+public class SearchBackendContext {
+	// TODO use a dedicated object for the error context instead of the backend
+	private final BackendImplementor<?> backend;
+
+	private final ElasticsearchWorkFactory workFactory;
+	private final MultiTenancyStrategy multiTenancyStrategy;
+
+	private final ElasticsearchWorkOrchestrator orchestrator;
+
+	private final DocumentReferenceHitExtractor documentReferenceHitExtractor;
+	private final ObjectHitExtractor objectHitExtractor;
+	private final DocumentReferenceProjectionHitExtractor documentReferenceProjectionHitExtractor;
+
+	public SearchBackendContext(BackendImplementor<?> backend,
+			ElasticsearchWorkFactory workFactory,
+			MultiTenancyStrategy multiTenancyStrategy,
+			ElasticsearchWorkOrchestrator orchestrator) {
+		this.backend = backend;
+		this.multiTenancyStrategy = multiTenancyStrategy;
+		this.workFactory = workFactory;
+		this.orchestrator = orchestrator;
+
+		this.documentReferenceHitExtractor = new DocumentReferenceHitExtractor( multiTenancyStrategy );
+		this.objectHitExtractor = new ObjectHitExtractor( multiTenancyStrategy );
+		this.documentReferenceProjectionHitExtractor = new DocumentReferenceProjectionHitExtractor( multiTenancyStrategy );
```

Comment:
They were previously hosted in an object that was created once per query, and there it would indeed have been a bad idea.
Here, though, the search backend context is created once per backend, and shared between all queries. So it's actually a good place to host "singletons" like this.

Reply:
Nice, thanks for the explanation.